search/search_repository.go

```diff
@@ -0,0 +1,357 @@
+package search
+
+import (
+	"fmt"
+
+	"golang.org/x/net/context"
+
+	"log"
+	"strconv"
+
+	"strings"
+
+	"regexp"
+
+	"net/url"
+
+	"github.com/almighty/almighty-core/app"
+	"github.com/almighty/almighty-core/models"
+	"github.com/asaskevich/govalidator"
+	"github.com/jinzhu/gorm"
+)
+
+const (
+	/*
+		- The SQL queries do a case-insensitive search.
+		- English words are normalized during search which means words like qualifying === qualify
+		- To disable the above normalization change "to_tsquery('english',$1)" to "to_tsquery($1)"
+		- Create GIN indexes : https://www.postgresql.org/docs/9.5/static/textsearch-tables.html#TEXTSEARCH-TABLES-INDEX
+		- To perform "LIKE" query we are appending ":*" to the search token
+
+	*/
+
+	// This SQL query is used when search is performed across workitem fields and workitem ID
+	WhereClauseForSearchByText = `setweight(to_tsvector('english',fields->>'system.title'),'B')||
```

Comment:
Very good observation @pranavgore09 ,
I agree with you that we need to have something like this, but not sure if we should be doing so right now because we are unsure about the implementation at a sql-query level about how to search multiple types. We need to figure out if we could go the single-query path , or multiple-query path.

Tracking it as part of #430